oci/oci.go

```diff
@@ -128,12 +128,25 @@ func (r *Runtime) CreateContainer(c *Container) error {
 	childPipe.Close()
 
 	// Wait to get container pid from conmon
-	// TODO(mrunalp): Add a timeout here
-	var si *syncInfo
-	if err := json.NewDecoder(parentPipe).Decode(&si); err != nil {
-		return fmt.Errorf("reading pid from init pipe: %v", err)
+	s := make(chan *syncInfo, 1)
+	e := make(chan error, 1)
+	go func() {
+		var si *syncInfo
+		if err = json.NewDecoder(parentPipe).Decode(&si); err != nil {
+			e <- fmt.Errorf("reading pid from init pipe: %v", err)
+			return
+		}
+		s <- si
+	}()
+
+	select {
+	case si := <-s:
+		logrus.Infof("Received container pid: %v", si.Pid)
+	case err = <-e:
+		return err
+	case <-time.After(3 * time.Second):
```

Comment:
we should avoid magic numbers, adds a config with default value or just a const if we agree on the default timeout value.